quickwit/quickwit-search/src/root.rs

```diff
@@ -1765,6 +1764,69 @@ pub fn jobs_to_fetch_docs_requests(
     Ok(fetch_docs_requests)
 }
 
+enum RootSearchMetricsStep {
+    Plan,
+    Exec { targeted_splits: usize },
+}
+
+/// Wrapper around the plan and search futures to track metrics.
+#[pin_project(PinnedDrop)]
+struct RootSearchMetricsFuture<F> {
+    #[pin]
+    tracked: F,
+    start: Instant,
+    step: RootSearchMetricsStep,
+    is_success: Option<bool>,
+}
+
+#[pinned_drop]
+impl<F> PinnedDrop for RootSearchMetricsFuture<F> {
+    fn drop(self: Pin<&mut Self>) {
+        let (targeted_splits, status) = match (&self.step, self.is_success) {
+            // is is a partial success, actual success is recorded during the search step
+            (RootSearchMetricsStep::Plan, Some(true)) => return,
+            (RootSearchMetricsStep::Plan, Some(false)) => (0, "plan-error"),
+            (RootSearchMetricsStep::Plan, None) => (0, "plan-cancelled"),
```

Comment:
These extra statuses seem valuable, but we should also try to avoid creating too many series. No strong opinion on whether we should have these or not.

(On that topic, I think `root_search_requests_total` is actually redundant because `root_search_request_duration_seconds` is a histogram and that creates a `_count` series)